components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/UserStoreConfigAdminService.java

```diff
@@ -330,118 +206,43 @@ public void editUserStore(UserStoreDTO userStoreDTO) throws IdentityUserStoreMgt
-            previousUserStoreConfigFile = Paths.get(tenantFilePath, String.valueOf(tenantId), previousFileName + ".xml");
-        }
-
-        if (!Files.exists(previousUserStoreConfigFile)) {
-            String errorMessage = "Cannot update user store domain name. Previous domain name " + previousDomainName +
-                    " does not exists.";
-            throw new IdentityUserStoreMgtException(errorMessage);
-        }
-
-        if (Files.exists(userStoreConfigFile)) {
-            String errorMessage = "Cannot update user store domain name. An user store already exists with new domain " +
-                    domainName + ".";
-            throw new IdentityUserStoreMgtException(errorMessage);
-        }
-
-        try {
-            // Run pre user-store name update listeners
-            List<UserStoreConfigListener> userStoreConfigListeners = UserStoreConfigListenersHolder.getInstance()
-                    .getUserStoreConfigListeners();
-            for (UserStoreConfigListener userStoreConfigListener : userStoreConfigListeners) {
-                userStoreConfigListener.onUserStoreNamePreUpdate(CarbonContext.getThreadLocalCarbonContext().getTenantId
-                        (), previousDomainName, domainName);
-            }
-
-            // Update persisted domain name
-            AbstractUserStoreManager userStoreManager = (AbstractUserStoreManager) CarbonContext.
-                    getThreadLocalCarbonContext().getUserRealm().getUserStoreManager();
-            userStoreManager.updatePersistedDomain(previousDomainName, domainName);
-            if (log.isDebugEnabled()) {
-                log.debug("Renamed persisted domain name from" + previousDomainName + " to " + domainName +
-                        " of tenant:" + tenantId + " from UM_DOMAIN.");
-            }
-
-        } catch (UserStoreException e) {
-            String errorMessage = "Error while updating user store domain : " + domainName;
-            log.error(errorMessage, e);
-            throw new IdentityUserStoreMgtException(errorMessage);
-        }
+    /**
+     * To get all the registered repository classes.
+     *
+     * @return repository classes
+     */
+    public String[] getRepositoryClasses() {
 
-        try {
-            Files.delete(previousUserStoreConfigFile);
-            writeUserMgtXMLFile(userStoreConfigFile, userStoreDTO, true);
-        } catch (IdentityUserStoreMgtException e) {
-            String errorMessage = e.getMessage();
-            log.error(errorMessage, e);
-            throw new IdentityUserStoreMgtException(errorMessage);
-        } catch (IOException e) {
-            log.info("Error when deleting previous configuration files " + previousUserStoreConfigFile);
-        }
+        Map<String, AbstractUserStoreDAOFactory> userStoreFactories = UserStoreConfigListenersHolder.getInstance().
+                getUserStoreDAOFactories();
+        Object[] repositoryArr = userStoreFactories.keySet().toArray();
```

Comment:
can't we just use as follows

userStoreFactories.keySet().toArray(new String[0]);